### PR TITLE
update to official rabbitmq client

### PR DESCRIPTION
**Pull request by AlexCuse**
I updated to latest watermill as well, this should close https://github.com/ThreeDotsLabs/watermill/issues/239

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,22 +1,22 @@
 module github.com/ThreeDotsLabs/watermill-amqp
 
-go 1.17
```
Comment (sagikazarmark) [inline go.mod:3]: I don't think you should remove this.

Comment (AlexCuse) [reply go.mod:3]: good catch @sagikazarmark I force pushed to undo.  Had to resolve a few mod issues on the CLI must have removed as part of that.  I just force pushed with the go 1.17 back in there thanks.

File: pkg/amqp/config.go [hunk 2]
```diff
@@ -2,12 +2,11 @@ package amqp
 
 import (
 	"crypto/tls"
+	"github.com/cenkalti/backoff/v3"
```
Comment (sagikazarmark) [inline pkg/amqp/config.go:5]: This should go back to its original place.

File: pkg/amqp/connection.go [hunk 3]
```diff
@@ -1,13 +1,13 @@
 package amqp
 
 import (
+	"github.com/cenkalti/backoff/v3"
```
Comment (sagikazarmark) [inline pkg/amqp/connection.go:4]: Same as above

File: pkg/amqp/config.go (hunk 2 above)
Comment (AlexCuse) [reply pkg/amqp/config.go]: these should be good

**Review by sagikazarmark — APPROVED** [review]

**Comment by roblaszczak** [comment]
LGTM! 

Thanks @AlexCuse for the PR and @sagikazarmark for the review!

**Pull request merged**